sql-plugin/src/main/scala/com/nvidia/spark/rapids/HostAlloc.scala

```diff
@@ -52,17 +54,35 @@ private class HostAlloc(nonPinnedLimit: Long) extends HostMemoryAllocator with L
     }
   }
 
+  private def reportHostAllocMetrics(metrics: GpuTaskMetrics): String = {
+    try {
+      val taskId = TaskContext.get().taskAttemptId()
+      val totalSize = metrics.getHostBytesAllocated
+      val maxSize = metrics.getMaxHostBytesAllocated
+      s"total size for task $taskId is $totalSize, max size is $maxSize"
+    } catch {
+      case _: NullPointerException =>
+        "allocated memory outside of a task context"
+    }
```

Comment:
I would add an explicit check for the null from `TaskContext.get()` rather than having this try-catch. The exception system should be used to handle unexpected errors at run time.

Reply:
:+1 to @jihoonson's suggestion. 

Plus, should we consider a different name for the function?  `reportHostAllocMetrics` implies that it does the reporting, but it's only generating the string.

Reply:
We can do a null check, or try the `Option` wrap pattern, we tend to do the option one.